src/main/java/com/github/javabdd/JFactory.java

```diff
@@ -1076,6 +1077,20 @@ BddCacheData copy() {
         }
     }
 
+    private static class BddCacheDataBI extends BddCacheData {
```

Comment:
Why not simply change the `BddCacheDataD` class? That class only seems to be used for satcount and pathcount. Adding this extra class comes with some extra complexity. For example, I'm not sure whether `BddCache#copy` now works correctly. This method now needs an extra case distinction for `BddCacheDataBI` entries.

Reply:
I fixed the `copy` method. I removed `BddCacheDataD`. So, it is kind of like renaming `BddCacheDataD` to `BddCacheDataBI` and changing the field from `double` to `BigInteger`.